### ISSUE TITLE
We need more CORS configurability

### ISSUE THREAD

**Issue by atruskie** [QutEcoacoustics/baw-server#677]
As the workbenches can be public APIs, we need random website to be able to access different API end points from a range of websites that we can't predetermine.

Our current CORS configurability is not good enough: https://github.com/QutEcoacoustics/baw-server/blob/40eb8002c1d10632b132b86b6fa94547ed637945/config/initializers/cors.rb#L8 . We can only allow whole origins.

So the plan is:

- Change CORS responses so that media GET requests allow cross origin
- Keep the CORS of /security routes restricted to the workbench client
-  Maybe: Change to http only auth cookeis and set Access-Control-Allow-Credentials

Basically, allow access for any public non-mutative routes/actions, but restrict security routes and any mutative actions for trusted origins.
